File: app/src/main/java/com/gaolei/famousfameproject/net/interceptor/HttpLoggingInterceptor.java
package com.gaolei.famousfameproject.net.interceptor;

import android.util.Log;


import com.gaolei.famousfameproject.BuildConfig;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * Created by liuhaiyang on 2016/9/6.
 */
public class HttpLoggingInterceptor implements Interceptor {
    @Override
    public Response intercept(Chain chain) throws IOException {
        Request request = chain.request();

        long t1 = System.nanoTime();

        if (BuildConfig.DEBUG) {
            Log.i("HTTP", String.format("----> %s %s on %s %n%s",
                    request.method(), request.url(), chain.connection(), request.headers()));
        }


        Response response = chain.proceed(request);

        long t2 = System.nanoTime();

        if (BuildConfig.DEBUG) {
            Log.i("HTTP", String.format("----> %s %s (%.1fms) %n%s", response.code(),
                    response.request().url(), (t2 - t1) / 1e6d, response.headers()));
        }


        return response;
    }
}

File: basemodule/src/main/java/com/android/base/thirdframe/retrofit/interceptor/HttpLoggingInterceptor.kt
package com.android.base.thirdframe.retrofit.interceptor


import com.android.base.utils.LogUtil

import java.io.IOException

import okhttp3.Interceptor
import okhttp3.Response


class HttpLoggingInterceptor : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        //这个chain里面包含了request和response，所以你要什么都可以从这里拿
        val request = chain.request()

        val t1 = System.nanoTime()//请求发起的时间
        LogUtil.d("HttpLoggingInterceptor", String.format("发送请求 %s on %s%n%s",
                request.url(), chain.connection(), request.headers()))

        val response = chain.proceed(request)

        val t2 = System.nanoTime()//收到响应的时间

        //这里不能直接使用response.body().string()的方式输出日志
        //因为response.body().string()之后，response中的流会被关闭，程序会报错，我们需要创建出一
        //个新的response给应用层处理
        val responseBody = response.peekBody((1024 * 1024).toLong())

        LogUtil.d("HttpLoggingInterceptor", String.format("接收响应: [%s] %n返回json:【%s】 %.1fms%n%s",
                response.request().url(),
                responseBody.string(),
                (t2 - t1) / 1e6,
                response.headers()))

        return response
    }
}

